### PR TITLE
Fix some NullReferences in Json files

### PR DESCRIPTION
**Pull request by KirillOsenkov**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/861121 - "System.NullReferenceException" when closing a solution while a json file is open & the solution doesn't close.

This is the MonoDevelop side of the fix for 861121. For non-C# files they could still be a part of WebEditorRoslynWorkspace, and not MonoDevelopWorkspace, and we need to gracefully deal with that.

### CODE REVIEW

**Comment by slluis** [comment]
@monojenkins backport master

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.WelcomePage/WelcomePageService.cs [hunk 1]
```diff
@@ -126,8 +126,15 @@ public static void ShowWelcomePage (bool animate = false)
...
+				
+				var rootWindow = (DefaultWorkbench)IdeApp.Workbench.RootWindow;
+				if (rootWindow.BottomBar is MonoDevelopStatusBar statusBar) {
+					statusBar.Visible = false;
+				}
+
+				if (rootWindow.DockFrame is Components.Docking.DockFrame dockFrame) {
+					dockFrame.AddOverlayWidget (welcomePage, animate);
+				}
 				welcomePage.GrabFocus ();
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.WelcomePage/WelcomePageService.cs]: What are the above changes for? they look unrelated to the bug

Comment (KirillOsenkov) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.WelcomePage/WelcomePageService.cs]: I basically fixed all the exceptions I saw when working with a .json file. With this PR there are no more exceptions along the codepaths I tested.

**Comment by slluis** [comment]
@monojenkins backport master
